src/mono/wasm/runtime/binding_support.js

```diff
@@ -931,7 +1065,325 @@ var BindingSupportLib = {
+				return "<null>";
+			return this.mono_wasm_get_type_aqn(typePtr);
+		},
+
+		_get_class_name: function (classPtr) {
+			if (!classPtr)
+				return "<null>";
+			return this.mono_wasm_get_type_name(this.mono_wasm_class_get_type(classPtr));
+		},
+
+		get_method_signature_info: function (typePtr, methodPtr) {
+			// MakeMarshalSignatureInfo is a managed method, so we'll get called
+			//  during the process of binding it
+			if (!this.make_marshal_signature_info)
+				return null;
+
+			if (!methodPtr)
+				throw new Error("Method ptr not provided");
+				
+			if (!this._method_signature_info_table)
+				this._method_signature_info_table = new Map ();
+			var result = this._method_signature_info_table.get (methodPtr);
+			var classMismatch = !!result && (result.typePtr !== typePtr);
+			if (!result) {
+				var typeName = this._get_type_name(typePtr);
+				var json = this.make_marshal_signature_info (typePtr, methodPtr);
+				if (!json)
+					throw new Error (`MakeMarshalSignatureInfo failed for type ${typeName}`);
+
+				var result = JSON.parse(json);
+				result.typePtr = typePtr;
+
+				if (classMismatch)
+					console.log("WARNING: Class ptr mismatch for signature info, so caching is disabled");
+				else
+					this._method_signature_info_table.set (methodPtr, result);
+			}
+			return result;
+		},
+
+		_compile_interchange_to_js: function (typePtr, boundConverter, js) {
+			if (!js)
+				return boundConverter;
+			
+			var closure = { 
+				MONO: MONO,
+				BINDING: this,
+				typePtr: typePtr, 
+				// (value) => filtered_value
+				temp_malloc: this._temp_malloc.bind(this)
+			};
+			var converterKey = boundConverter.name || "boundConverter";
+			closure[converterKey] = boundConverter;
+
+			var filterExpression = this._create_named_function(
+				"interchange_to_js_filter_for_type" + typePtr,
+				["value"], js, closure
+			);
+			closure.filter = filterExpression;
+			
+			var bodyJs = `var value = ${converterKey}(js_value), filteredValue = filter(value);\r\n` +
+				"return filteredValue;";
+			var functionName = "interchange_to_js_for_type" + typePtr;
+			var result = this._create_named_function(
+				functionName, ["js_value"], bodyJs, closure
+			);
+
+			return result;
+		},
+
+		_pre_filter_date: function (value) {
+			switch (typeof (value)) {
+				case "number":
+					return value;
+				default:
+					if (value instanceof Date)
+						return value.valueOf();
+					else
+						throw new Error("Value must be a number (msecs since unix epoch), or a Date");
+			}
+		},
+
+		_get_custom_marshaler_info_for_type: function (typePtr) {
+			if (!typePtr)
+				return null;
+			if (!MONO._custom_marshaler_name_table)
+				return null;
+
+			if (!this._custom_marshaler_info_cache)
+				this._custom_marshaler_info_cache = new Map ();
+			
+			var result;
+			if (!this._custom_marshaler_info_cache.has (typePtr)) {
+				var aqn = this._get_type_aqn (typePtr);
+				if (!aqn.startsWith("System.Object, System.Private.CoreLib, ")) {
```

Comment:
Why do you need to special case object here?

Reply:
Object needs to go through the old 'create a handle and track the object' flow under all circumstances, so this suppresses any erroneous attempt to use a custom marshaler for it or generate warnings if it's missing one

Reply:
I don't think it should be done this way. I think for now we can ignore this because we'll most likely have some problem with other core types too.

Reply:
There are spurious warnings in the log for all applications right now without this

Reply:
Where are they coming from? Why would object marshalling even hit this code path?

Reply:
Any object being unboxed first has to have its type checked and looked up in the converter table, that will include System.Object